src/Stubs.cs

```diff
@@ -1,23 +1,38 @@
 using SpacetimeDB.ClientApi;
 
+using System;
+
 namespace SpacetimeDB
 {
     public interface IReducerArgsBase : BSATN.IStructuralReadWrite
     {
         string ReducerName { get; }
     }
 
-    public abstract class ReducerEventBase
+    public abstract class DbContext<DbView, ReducerView> : DbContext<DbView>
+        where DbView : class, new()
+        where ReducerView : class, new()
+    {
+        public ReducerView Reducers;
+
+        public DbContext(DbView db, ReducerView reducers) : base(db) => Reducers = reducers;
+    }
+
+    public interface IEventContext {
+        bool InvokeHandler();
+    }
+
+    public abstract class EventContextBase<RemoteTables, RemoteReducers> : DbContext<RemoteTables, RemoteReducers>, IEventContext

```

Comment:
Just in case - this doesn't seem to match the structure described in the "SDK callbacks" section of the proposal - are those changes left for a separate PR?